### ISSUE TITLE
MrVI - minimal number of genes

### ISSUE THREAD

**Issue by dansteiert** [scverse/scvi-tools#3347]
Hey,
I am running MrVI on two datasets - a scRNAseq dataset (~30k genes) and a protein dataset (~40 genes).
With the scRNAseq Dataset everything is working very well!
The 40gene dataset does not unfortunately.
-> Training actually finishes (over multiple iterrations)
-> looking into the losses (e.g. model.history()) are all nan.

Is there a lower threshold for number of genes the model wants to learn on - e.g. first layer has higher dimensions as the input layer?

could these be datatype issues?

or do I need to pass specific parameters to the model - currently I am running this with no additional arguments
```
MRVI.setup_anndata(adata, sample_key=sample_key, batch_key=batch_key)
model = MRVI(adata)
```
Very thankful for your insights!

**Comment by ori-kron-wis** [comment]
Can you show a snippet of your model history loss numbers to show if it was from the beginning or just in the last epoch? (How many epochs did he go with loss = Nan)


There is no minimum number of genes needed, but you can check for some default training parameters in the code:
```
    n_latent: int = 30
    n_latent_u: int = 10
    encoder_n_hidden: int = 128
    encoder_n_layers: int = 2
    z_u_prior: bool = True
    z_u_prior_scale: float = 0.0
    u_prior_scale: float = 0.0
    u_prior_mixture: bool = True
    u_prior_mixture_k: int = 20
```
```
DEFAULT_TRAIN_KWARGS = {
    "max_epochs": 100,
    "early_stopping": True,
    "early_stopping_patience": 15,
    "check_val_every_n_epoch": 1,
    "batch_size": 256,
    "train_size": 0.9,
    "plan_kwargs": {
        "lr": 2e-3,
        "n_epochs_kl_warmup": 20,
        "max_norm": 40,
        "eps": 1e-8,
        "weight_decay": 1e-8,
    },
}
```
max_norm=40 seems suspicious, but not sure it will affect you.

I tried a dummy example with 40 genes and it worked with default settings, so it may relate to your specific data.

You can also try to run our autotune to find set of training parameters or model parameters that will work for your case.

**Comment by dansteiert** [comment]
To the model history, it is NaN everywhere from the first epoch.

Thank you I will have a look into the autotune function and see If I can find something there, to make it work for this dataset

**Comment by PierreBoyeau** [comment]
Thanks a lot for your interest in this model. 
MrVI has not been tested on data with that few features, let alone on protein data. For this reason, I would advise against using MrVI on such data. To work with protein data, I'd probably 1. make sure that the protein data corresponds to raw and untransformed count data (otherwise, training and inference will definitely raise NaN errors), and 2. adjust MrVI likelihood taking inspiration from TOTALVI, for instance.
I am closing this, but feel free to reopen if you have follow-up questions!
